test/e2e/packetcapture_test.go

```diff
@@ -0,0 +1,728 @@
+	// #nosec G101
+	pcSecretName     = "antrea-packetcapture-fileserver-auth"
+	tcpServerPodName = "tcp-server"
+	pcToolboxPodName = "toolbox"
+	udpServerPodName = "udp-server"
+	nonExistPodName  = "non-existing-pod"
+
+	tcpProto  = intstr.FromString("TCP")
+	icmpProto = intstr.FromString("ICMP")
+	udpProto  = intstr.FromString("UDP")
+
+	testServerPort   int32 = 80
+	testNonExistPort int32 = 8085
+
+	pcTimeoutReason = "PacketCapture timeout"
+	pcShortTimeout  = uint16(5)
+)
+
+type pcTestCase struct {
+	name           string
+	pc             *crdv1alpha1.PacketCapture
+	expectedStatus crdv1alpha1.PacketCaptureStatus
+
+	// required IP version, skip if not match.
+	ipVersion int
+	// Source Pod to run ping for live-traffic PacketCapture.
+	srcPod string
+}
+
+func genSFTPService() *v1.Service {
+	selector := map[string]string{"app": "sftp"}
+	return &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "sftp",
+			Labels: selector,
+		},
+		Spec: v1.ServiceSpec{
+			Type:     v1.ServiceTypeNodePort,
+			Selector: selector,
+			Ports: []v1.ServicePort{
+				{
+					Port:       22,
+					TargetPort: intstr.FromInt32(22),
+					NodePort:   30010,
+				},
+			},
+		},
+	}
+}
+
+func genSFTPDeployment() *appsv1.Deployment {
+	replicas := int32(1)
+	selector := map[string]string{"app": "sftp"}
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "sftp",
+			Labels: selector,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: selector,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   "sftp",
+					Labels: selector,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:            "sftp",
+							Image:           "antrea/sftp",
+							ImagePullPolicy: v1.PullIfNotPresent,
+							Args:            []string{"foo:pass:::upload"},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createUDPServerPod(name string, ns string, portNum int32, serverNode string) error {
+	port := v1.ContainerPort{Name: fmt.Sprintf("port-%d", portNum), ContainerPort: portNum}
+	return NewPodBuilder(name, ns, agnhostImage).
+		OnNode(serverNode).
+		WithContainerName("agnhost").
+		WithArgs([]string{"serve-hostname", "--udp", "--http=false", "--port", fmt.Sprint(portNum)}).
+		WithPorts([]v1.ContainerPort{port}).
+		Create(testData)
+}
+
+// TestPacketCapture is the top-level test which contains all subtests for
+// PacketCapture related test cases, so they can share setup, teardown.
+func TestPacketCapture(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	var previousAgentPacketCaptureEnableState bool
+	ac := func(config *agentconfig.AgentConfig) {
+		previousAgentPacketCaptureEnableState = config.FeatureGates[string(features.PacketCapture)]
+		config.FeatureGates[string(features.PacketCapture)] = true
+	}
```

Comment:
I don't believe we do this in e2e tests anymore (change the value of a FeatureGate) unless there is a good reason. Instead, we skip the test if the FeatureGate is not enabled. Alpha features are tested by the `test-e2e-encap-all-features-enabled` job, for which we enable all alpha features.

Reply:
good to know. removed.